src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Commands.CreateApprenticeFeedback
+{
+    public class CreateApprenticeFeedbackHandler : IRequestHandler<CreateApprenticeFeedbackCommand, CreateApprenticeFeedbackResponse>
+    {
+        public readonly IApprenticeFeedbackRepository _apprenticeFeedbackRepository;
+
+        public CreateApprenticeFeedbackHandler(IApprenticeFeedbackRepository apprenticeFeedbackRepository)
+        {
+            _apprenticeFeedbackRepository = apprenticeFeedbackRepository;
+        }
+
+
+        public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedbackCommand request, CancellationToken cancellationToken)
+        {
+
+            // Needs to retrieve apprentice feedback target based on apprentice id which should be in request
+            // May not return a single record. ( currently we assume it returns one )
+            var apprenticeFeedbackTarget = await _apprenticeFeedbackRepository.GetApprenticeFeedbackTargetById(request.ApprenticeId);
```

Comment:
Aligning with the previous comment, if the ApprenticeFeedbackTarget is supplied, this should make this line easier as it is the unique key over the apprentice Id

Reply:
So I need to changes the method to take the ApprenticeFeedbackTargetId rather than ApprenticeId? (This would also solve our problem of this method potentially returning more than one record)